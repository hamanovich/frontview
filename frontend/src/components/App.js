import React, { Fragment, lazy } from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import HomePage from './HomePage';
import NotFound from './NotFound';
import Header from './layout/Header';
import Footer from './layout/Footer';
import FlashList from './flash/FlashList';
import FaqButton from './shared/FaqButton';

import { isLoggedIn, User, Waiting } from '../utils/helpers';

const LoginPage = lazy(() => import('./LoginPage' /* webpackChunkName: "Login" */));
const SignupPage = lazy(() => import('./SignupPage' /* webpackChunkName: "Signup" */));
const AccountPage = lazy(() => import('./AccountPage' /* webpackChunkName: "AccountPage" */));
const InterviewPage = lazy(() => import('./InterviewPage' /* webpackChunkName: "InterviewPage" */));
const QuestionsPage = lazy(() => import('./QuestionsPage' /* webpackChunkName: "Questions" */));
const CommentsAuthorPage = lazy(() =>
  import('./Comment/CommentsAuthorPage' /* webpackChunkName: "CommentsAuthor" */),
);
const ConfirmationPage = lazy(() =>
  import('./ConfirmationPage' /* webpackChunkName: "Confirmation" */),
);
const FaqPage = lazy(() => import('./FaqPage' /* webpackChunkName: "FAQ" */));

const App = () => (
  <Fragment>
    <Helmet>
      <title>Frontview: Interview for Everyone</title>
    </Helmet>
    <Header />
    <main id="main">
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/login" component={isLoggedIn(Waiting(LoginPage))} />
        <Route path="/signup" component={isLoggedIn(Waiting(SignupPage))} />
        <Route path="/me" component={User(Waiting(AccountPage))} />
        <Route path="/questions" component={Waiting(QuestionsPage)} />
        <Route path="/interview" component={User(Waiting(InterviewPage))} />
        <Route path="/comments/:username" component={User(Waiting(CommentsAuthorPage))} />
        <Route path="/confirmation/:token" component={Waiting(ConfirmationPage)} />
        <Route path="/faq" component={Waiting(FaqPage)} />
        <Route component={NotFound} />
      </Switch>
    </main>
    <Footer />
    <FlashList />
    <FaqButton />
  </Fragment>
);

export default App;
