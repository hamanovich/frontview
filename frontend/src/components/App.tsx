import React, { Fragment, lazy } from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import ScrollToTop from 'react-scroll-up';
import FontAwesome from 'react-fontawesome';

import Button from 'react-bootstrap/Button';

import HomePage from './HomePage';
import NotFound from './NotFound';
import Header from './layout/Header';
import Footer from './layout/Footer';
import FlashList from './flash/FlashList';
import FaqButton from './shared/FaqButton';

import { isLoggedIn, Customer, Waiting } from '../utils/helpers';
import ScrollToTopRoute from '../utils/ScrollToTopRoute';

const LoginPage = lazy(() =>
  import('./LoginPage' /* webpackChunkName: "Login" */),
);
const SignupPage = lazy(() =>
  import('./SignupPage' /* webpackChunkName: "Signup" */),
);
const AccountPage = lazy(() =>
  import('./AccountPage' /* webpackChunkName: "AccountPage" */),
);
const AboutPage = lazy(() =>
  import('./AboutPage' /* webpackChunkName: "AboutPage" */),
);
const InterviewPage = lazy(() =>
  import('./InterviewPage' /* webpackChunkName: "InterviewPage" */),
);
const QuestionsPage = lazy(() =>
  import('./QuestionsPage' /* webpackChunkName: "Questions" */),
);
const CommentsAuthorPage = lazy(() =>
  import('./Comment/CommentsAuthorPage' /* webpackChunkName: "CommentsAuthor" */),
);
const ConfirmationPage = lazy(() =>
  import('./ConfirmationPage' /* webpackChunkName: "Confirmation" */),
);

const App = () => (
  <Fragment>
    <Helmet>
      <title>Frontview: Interview for Everyone</title>
    </Helmet>
    <Header />
    <main id="main">
      <ScrollToTopRoute />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/login" component={isLoggedIn(Waiting(LoginPage))} />
        <Route path="/signup" component={isLoggedIn(Waiting(SignupPage))} />
        <Route path="/me" component={Customer(Waiting(AccountPage))} />
        <Route path="/questions" component={Waiting(QuestionsPage)} />
        <Route path="/interview" component={Customer(Waiting(InterviewPage))} />
        <Route
          path="/comments/:username"
          component={Customer(Waiting(CommentsAuthorPage))}
        />
        <Route
          path="/confirmation/:token"
          component={Waiting(ConfirmationPage)}
        />
        <Route path="/about" component={Waiting(AboutPage)} />
        <Route component={NotFound} />
      </Switch>
    </main>
    <Footer />
    <FlashList />
    <FaqButton />
    <ScrollToTop showUnder={160} style={{ bottom: 16, right: 16 }}>
      <Button variant="warning">
        <FontAwesome name="chevron-up" />
      </Button>
    </ScrollToTop>
  </Fragment>
);

export default App;
