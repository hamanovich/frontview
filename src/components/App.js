import React from 'react';
import { Switch, Route } from 'react-router-dom';
import styled from 'styled-components';

import Grid from 'react-bootstrap/lib/Grid';

import HomePage from './HomePage';
import NotFound from './NotFound';
import Header from './layout/Header';
import Footer from './layout/Footer';
import FlashList from './flash/FlashList';
import ConfirmationPage from './ConfirmationPage';
import InterviewPage from './InteviewPage';
import LoginPage from './LoginPage';
import SignupPage from './SignupPage';
import AccountPage from './AccountPage';
import QuestionsPage from './QuestionsPage';
import CommentsAuthorPage from './Comment/CommentsAuthorPage';

import { isLoggedIn, User } from '../utils/helpers';

const Main = styled.main`
  padding-bottom: 30px;
`;

const App = () => (
  <div>
    <Header />
    <Grid>
      <FlashList />
      <Main>
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/login" component={isLoggedIn(LoginPage)} />
          <Route path="/signup" component={isLoggedIn(SignupPage)} />
          <Route path="/me" component={User(AccountPage)} />
          <Route path="/questions" component={User(QuestionsPage)} />
          <Route path="/interview" component={User(InterviewPage)} />
          <Route path="/comments/:username" component={User(CommentsAuthorPage)} />
          <Route path="/confirmation/:token" component={ConfirmationPage} />
          <Route component={NotFound} />
        </Switch>
      </Main>
    </Grid>
    <Footer />
  </div>
);

export default App;