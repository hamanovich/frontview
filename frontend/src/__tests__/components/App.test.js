import React from 'react';
import { shallow } from 'enzyme';

import App from '../../components/App';

describe('<App/>', () => {
  it('renders <App /> component', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });
});
