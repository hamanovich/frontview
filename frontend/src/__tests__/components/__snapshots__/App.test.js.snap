// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders <App /> component 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Frontview: Interview for Everyone
    </title>
  </HelmetWrapper>
  <Connect(Header) />
  <main
    id="main"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/me"
      />
      <Route
        component={[Function]}
        path="/questions"
      />
      <Route
        component={[Function]}
        path="/interview"
      />
      <Route
        component={[Function]}
        path="/comments/:username"
      />
      <Route
        component={[Function]}
        path="/confirmation/:token"
      />
      <Route
        component={[Function]}
        path="/faq"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
  <Footer />
  <Connect(FlashList) />
  <FaqButton />
</Fragment>
`;
